composeApp/src/androidMain/kotlin/org/ooni/probe/background/RunWorkerManager.kt

```diff
@@ -0,0 +1,64 @@
+package org.ooni.probe.background
+
+import androidx.work.Constraints
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.ExistingWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.OutOfQuotaPolicy
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import org.ooni.probe.data.models.AutoRunParameters
+import org.ooni.probe.data.models.RunSpecification
+import java.util.concurrent.TimeUnit
+
+class RunWorkerManager(
+    private val workManager: WorkManager,
+    private val backgroundDispatcher: CoroutineDispatcher,
+) {
+    fun startSingleRun(spec: RunSpecification) {
+        workManager
+            .enqueueUniqueWork(
+                RUN_UNIQUE_WORKER_NAME,
+                ExistingWorkPolicy.KEEP,
+                OneTimeWorkRequestBuilder<RunWorker>()
+                    .setInputData(RunWorker.buildWorkData(spec))
+                    .setExpedited(OutOfQuotaPolicy.RUN_AS_NON_EXPEDITED_WORK_REQUEST)
+                    .build(),
+            )
+    }
+
+    suspend fun configureAutoRun(params: AutoRunParameters) {
+        withContext(backgroundDispatcher) {
+            if (params !is AutoRunParameters.Enabled) {
+                workManager.cancelUniqueWork(AUTO_RUN_UNIQUE_WORKER_NAME)
+                return@withContext
+            }
+
+            val request = PeriodicWorkRequestBuilder<RunWorker>(1, TimeUnit.HOURS)
```

Comment:
There is some constraint if the device VPN is turned on. We should add a TODO to get the teams opinion on this.

Measurements over a VPN usually measures the state of your VPN and not the actual network.